### ISSUE TITLE
Floating emoji paralax component

### ISSUE THREAD

**Issue by GarlandKey** [thinkbig-project/thinkbig#22]
On the front page, there will be brain emojis that float from the bottom of the screen to the top. They will populate randomly and in various sizes. The larger the emoji is, the faster it will move (to create a parallax effect). As they move upward, they will also randomly shift from side to side.

### The component should do the following:
1. Take up the entire view-port.
2. The images should populate randomly, within a given scale of time.
3. The images should have a range of scale that is chosen randomly.
4. The images should have a movement speed tied to that randomly chosen size.
5. The images should always move upward at its given speed.
6. As the images move upward, they should slowly and randomly bounce from side to side. There should be a range of motion so that they don't start drifting sideways too much.

**Sample image:**

![brain_demo](https://github.com/thinkbig-project/thinkbig/assets/3108516/ea9655f7-f121-4996-81d2-68dfeee7eb97)

**Resources:**
[React - Creating your first component](https://react.dev/learn/your-first-component)
[TailwindCSS Docs](https://tailwindcss.com/docs/aspect-ratio)
[Creating custom animations in TailwindCSS](https://blog.logrocket.com/creating-custom-animations-tailwind-css/)

**Comment by vasfvitor** [comment]
I won't be able to finish this until Saturday. If anyone want to continue feel free. I'll leave a note about how I've started doing it.

My approach is a animation that goes 0% 100vh to 100% 0vh (or -100vh, don't recall). 

A span element with the emoji. A div that wraps it. fixed z-20 bottom-0

Then I believe we can't use only tailwind classes to get the randomness we need, even with one off classes. So I was trying with a dynamic style, left position, animation delay and duration, and font size.

I haven't tested for performance and so far to achieve multiple emojis it would have to place multiple repeated components. A possible issue is regarding the z position, so the animation is on the background, but not bwhind everytehind and not in front of content. So i guess a z index prop would be good.

Im away from my PC, so I can't share| the code rn. Hope it helps. I'd love to see a better implementation tho

**Comment by GarlandKey** [comment]
@vasfvitor Would it help to break this up into smaller issues? For example:
Issue 1: Make image component that moves from bottom of viewport to the top.
Issue 2: Make image component move randomly from side to side.
Issue 3: Make image component appear at a random size.
Issue 4. Make a full viewport component that populates the viewport with image component randomly.

**Comment by vasfvitor** [comment]
This could be a possibility, but then it should be clear what's the final goal so it is done with it in mind, I mean, I can see a case where, for example in the last issue we find out that the implementation needs to be refactored to allow such feat (not in this specific case, but in others), can't it instead be divided in tasks? 

Btw, have you done this or seen something similar? this way we can adapt on it to see how this can go.

**Comment by vasfvitor** [comment]
this is what is done so far:
```tsx Emoji.tsx
'use client'
import { useState } from 'react';
import * as React from 'react';
import './styles.css'


interface ParallaxProps {
    emoji: string;
}

const randomNumber = (min: number, max: number) => {
    return Math.floor(Math.random()
        * (max - min + 1)) + min;
};

const Parallax: React.FC<ParallaxProps> = ({ emoji }) => {
    const [seed, setSeed] = useState(randomNumber(1, 10));
    const [left, setLeft] = useState(seed * 10);
    const [size, setSize] = useState(seed * 12);

    // duration in ms
    const [time, setTime] = useState((12 - seed) * 500);

    // delay in ms
    const [delay, setDelay] = useState(Math.floor(Math.random() * 100));

    const AnimationEnd = () => {
        setSeed(randomNumber(5, 10))
    };
    return (<div style={{
        left: `${left}%`,
        fontSize: `${size}px`,
        animation: `rise ${time}ms ease-in ${delay}ms infinite`,
    }} onAnimationEnd={AnimationEnd} className="z-20 fixed bottom-0">
        <span>{emoji}</span>
    </div>
    );
};

export default Parallax;

```

```css styles.css
keyframes rise {
  0% {
    transform: translateY(100vh);
  }
  100% {
    transform: translateY(-100vh);
  }
}
```

Placing 7 components in the page. Btw, is there a way to instance multiple components? 
As you can see whats left is to randomize it's state and the maybe create a portal to place it to the body
Preview:
![firefox_uFJBOGTdho](https://github.com/thinkbig-project/thinkbig/assets/61759797/e91e1c69-ea67-4a21-8468-0b034bd799aa)

**Comment by SupaMega24** [comment]
We could also go with something a bit simpler, then come back to later once the site is fully functional. 

If it's alright, I'll play around with some ideas over the weekend.

**Comment by SupaMega24** [comment]
Couldn't get it to work.

**Comment by vasfvitor** [comment]
@SupaMega24 you couldn't get my example to work or overall? I'll push the work I've done to the branch

**Comment by SupaMega24** [comment]
@vasfvitor I didn't try yours. Honestly, I don't have a lot of experience yet with Next and React. I can make some basic components, but I'm still learning. I just figured I'd give it a shot.

**Comment by SabAlej** [comment]
Hi @GarlandKey, hope you're doing great! I don't know if someone is working on this, if not, I'd like to work on it, can you assign me if it's the case please? 😸🌻Have an awesome day!

**Comment by SabAlej** [comment]
Hello, me again 🙈 sorry about the PR #44, i made a mistake with the branches and couldn't edit. So i closed it and opened a new one, PR #45, the only problem is that only the first one was assigned automatically. I hope it helps, let me know if i have to change something. Thank you!🌻

**Comment by SabAlej** [comment]
Thank you very much @vasfvitor for linking the rigth PR. You are fast!!! 😎

**Comment by GarlandKey** [comment]
@vasfvitor That looks great!

@SabAlej @vasfvitor @SupaMega24 Sorry for the late response. I got covid after Christmas and have been afk this whole time. Getting back into the swing of things now. Let me catch up on these threads and I'll send everyone another message.

**Comment by GarlandKey** [comment]
Okay, @vasfvitor, please re-assign this to yourself and submit it for a pull request.
@SabAlej I will create another issue for completing this issue once the pull request is merged.
